hw/dv/dpi/common/tcp_server/tcp_server.c

```diff
@@ -21,7 +21,11 @@
 /**
  * Simple buffer for passing data between TCP sockets and DPI modules
  */
+#ifdef __cplusplus
```

Comment:
Do you need this `ifdef`? can you not just make this a:
`static const int BUFSIZE_BYTE = 256;`?

Reply:
static const int BUFSIZE_BYTE = 256 results in:
```
opentitan/hw/dv/dpi/common/tcp_server/tcp_server.c:29:8: error: variably modified 'buf' at file scope
   29 |   char buf[BUFSIZE_BYTE];
```
I could drop the ifdef if we do
```
#define BUFSIZE_BYTE 256
```

I'm fine without the ifdef if we #define it. Is there was a C++ reason to prefer static const int over #define INT ?

Reply:
ah right this is C not C++.  the #define is good in that case.

The reason comes from our [style guide](https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros), but its ok in this case b/c of the given circumstances (compiling with both C and C++).